admin/data/sql/migrations/20190210134733.do.create-view-schools-average.sql

```diff
@@ -0,0 +1,31 @@
+CREATE VIEW [mtc_admin].[vewSchoolsAverage] AS
+
+SELECT
+  p.school_id,
+  CAST(SUM(
+      CASE
+      WHEN (pa.id IS NOT NULL) THEN 0
+      WHEN (latestPupilCheck.mark IS NOT NULL) THEN latestPupilCheck.mark
+      ELSE 0
+      END
+  ) AS DECIMAL(5,2)) / COUNT(p.id) AS [schoolAverage]
```

Comment:
according to [the documentation ](https://docs.microsoft.com/en-us/sql/t-sql/functions/sum-transact-sql?view=sql-server-2017)SUM will adapt and return the correct datatype based on the inputs.  have you tested to see if it dynamically returns decimal without an explicit cast?

Reply:
@GuyHarwood - it returns a decimal as you surmise, but the cast will limit the number of decimal places to two.  So the choice is to either keep the cast and limit the decimal places to two, or lose it and have lots of decimal places.

Reply:
@GuyHarwood I tested this without the cast and it does not return a decimal.